File: driver-service/src/main/java/com/dulfinne/taxi/driverservice/kafka/config/KafkaProperties.java
package com.dulfinne.taxi.driverservice.kafka.config;

import lombok.Data;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

@Component
@Data
public class KafkaProperties {

  @Value("${spring.kafka.bootstrap-servers}")
  private String bootstrapServers;

  @Value("${spring.kafka.schema-registry.url}")
  private String schemaRegistryUrl;

  @Value("${spring.kafka.groups.driver-rating}")
  private String driverRatingGroup;

  @Value("${spring.kafka.topics.rate-driver}")
  private String rateDriverTopic;

  @Value("${spring.kafka.consumers-number}")
  private Integer consumersNumber;
}

File: ride-service/src/main/kotlin/com/dulfinne/taxi/rideservice/kafka/config/KafkaProperties.kt
package com.dulfinne.taxi.rideservice.kafka.config

import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

@Component
class KafkaProperties {

    @Value("\${spring.kafka.bootstrap-servers}")
    lateinit var bootstrapServers: String

    @Value("\${spring.kafka.schema-registry.url}")
    lateinit var schemaRegistryUrl: String

    @Value("\${spring.kafka.topics.rate-passenger}")
    lateinit var ratePassengerTopic: String

    @Value("\${spring.kafka.topics.rate-driver}")
    lateinit var rateDriverTopic: String

    @Value("\${spring.kafka.topics.promocode-usage}")
    lateinit var promocodeUsageTopic: String

    @Value("\${spring.kafka.topics.ride-payment}")
    lateinit var ridePaymentTopic: String

    @Value("\${spring.kafka.partition-number}")
    var partitionNumber: Int = 1

    @Value("\${spring.kafka.replicas-number}")
    var replicasNumber: Int = 1
}